test/testinfra/MUXTestInfra/TestAppInstallHelper.cs

```diff
@@ -137,13 +135,58 @@ FileInfo FirstFileWithExtension(params string[] extensions)
                 }
                 else
                 {
-                    Log.Comment("Test Appx Package was not found in {0}.", deploymentDir);
+                    Log.Error("Test Appx Package was not found in {0}.", deploymentDir);
+                    return false;
                 }
 
                 TestAppxInstalled.Add(packageFamilyName);
             }
+
+            return true;
         }
-        
+
+        /// <summary>
+        /// Installs the unit test app from a package
+        /// </summary>
+        public static bool InstallTestAppFromDirectoryIfNeeded(string testAppDirectory, string packageFamilyName)
+        {
+            if (!TestAppxInstalled.Contains(packageFamilyName))
+            {
+                var createAppxDirectoryScript = Path.Combine(testAppDirectory, "CreateAppxDirectory.ps1");
+
+                if (File.Exists(createAppxDirectoryScript))
+                {
+                    // To install from a directory, we have a script that does two things: first, it runs an MSBuild script
```

Comment:
Instead of invoking MSBuild from the test code (which is a bit topsy-turvy) can we make the CreateAppxPackage Target run as part of the main build?